service/src/main/java/io/camunda/service/license/CamundaLicense.java

```diff
@@ -7,41 +7,50 @@
  */
 package io.camunda.service.license;
 
-import org.apache.commons.lang3.StringUtils;
+import org.camunda.bpm.licensecheck.InvalidLicenseException;
+import org.camunda.bpm.licensecheck.LicenseKey;
+import org.camunda.bpm.licensecheck.LicenseKeyImpl;
 
 public class CamundaLicense {
 
-  public static final String LICENSE_ENV_VAR_KEY = "CAMUNDA_LICENSE_KEY";
-  public static String licenseStr;
-  private final EnvironmentVariableReader environmentVariableReader;
+  public String license;
+  public boolean isValid;
+  private volatile boolean isInitialized;
 
-  public CamundaLicense() {
-    environmentVariableReader = new EnvironmentVariableReader();
+  public CamundaLicense(final String license) {
+    this.license = license;
   }
 
-  /**
-   * Used for testing. Test can mock EnvironmentVariableReader to test out different environment
-   * variable values
-   */
-  public CamundaLicense(final EnvironmentVariableReader environmentVariableReader) {
-    this.environmentVariableReader = environmentVariableReader;
-  }
+  // for testing
+  public CamundaLicense() {}
```

Comment:
you can make package visible constructor and annotate it with `@VisibleForTesting`

Reply:
Done! I didn't know this annotation was available